ath79: add support for KuWfi CPE830(D) / YunCore CPE830(D)

grosjo opened the pull request:
Somehow, I need this to move forward

People can critize and blame as much as they want,  but if nothing is done, those poeple better keep quiet

ambassador86 commented:
Isn't this a copy of the recently closed #2354 ?

adrianschmutzler commented:
@ambassador86 It is. As you might see in the previous one, there had been an unpleasant discussion about the PR and it eventually was closed by a core-developer.
I expect this to be closed soon, too, as I do not see any steps forward from looking at it (briefly).

ambassador86 commented:
Hello @adrianschmutzler,
the commit id is still the same so I expect it is the same content.

grosjo commented:
It is the same, but nobody tells what is wrong with it, except multiple insults

981213 commented on target/linux/ath79/dts/qca9531_yuncore_cpe830.dts:
```diff
@@ -0,0 +1,122 @@
+	model = "YunCore/KuWfi CPE830(D)";
+
+	chosen {
+		bootargs = "console=ttyS0,115200 rootfstype=squashfs,jffs2 noinitrd";
```
bootargs in chosen node will override what u-boot cmdline tells kernel so u-boot can't be messing up with kernel cmdline if you have a bootargs defined here. I'm not sure whether "noinitrd" is required but rootfstype should definitely be dropped because it's already defined in target/linux/ath79/config-4.19

> CONFIG_CMDLINE="rootfstype=squashfs,jffs2"

981213 reviewed (changes requested):
Hi!
I'm too lazy to go through previous  PRs for this router so here's my own review below:

psyborg55 commented:
Invalid pull request

981213 commented:
Well. I tried to read discussions in previous PRs and here's what I've got:
The main issue is that this PR adds support for two devices, claiming CPE830 is the same as CPE830D while @grosjo only has one of them and can't verify the firmware on the other. If that's the case I'd say that this PR indeed can't be accepted without a proper test on the other. (Or you can just change device titles so that only device support for the tested one is added.)

@psyborg55 

> Invalid pull request

Saying it's invalid again and again without providing a reason isn't helpful at all.

CodeFetch commented on target/linux/ath79/dts/qca9531_yuncore_cpe830.dts:
```diff
@@ -0,0 +1,122 @@
+	model = "YunCore/KuWfi CPE830(D)";
+
+	chosen {
+		bootargs = "console=ttyS0,115200 rootfstype=squashfs,jffs2 noinitrd";
```
@981213 I think the whole line should be dropped as the router does not have a UART header at all.

981213 replied:
@CodeFetch
If the entire line is dropped, u-boot cmdline will be used and that will cause problems if vendor decided to pass arguments like rootfstype.

CodeFetch replied:
@981213 Thank you for the clarification. I see CONFIG_CMDLINE_OVERRIDE is undefined and there is no default override in DTS... I think there was a discussion on the mailing list a while back with the conclusion that OpenWrt should never retain the bootcmds of the vendor's bootloader... I think grosjo should provide a dump of the U-Boot environment partition of an unmodified router or use uboot-envtools to make clear why overriding the bootcmd is needed. I can't imagine any reason.

grosjo replied:
fixed

grosjo commented on target/linux/ath79/image/common-yuncore.mk:
```diff
@@ -0,0 +1,11 @@
+	( \
+		echo -n -e "YUNCOREsetenv bootcmd \"bootm 0x9f050000 || bootm 0x9fe80000\" && " \
+			"setenv resetcmd \"erase 0x9f050000 +0xfa0000 && cp.b 0x800600c0 0x9f050000 0xfa0000\" && " \
+			"setenv bootargs \"console=ttyS0,115200 rootfstype=squashfs,jffs2 noinitrd\" && " \
```
Resetcmd is used by custom uboot of the device.

grosjo replied:
I removed the bootargs , and kept the one in DTS

CodeFetch replied:
No. To clarify:

The resetcmd saved in the U-Boot environment variable is executed when there is no resetcmd defined at the beginning of the image. By default there is no resetcmd defined. As we define a resetcmd in the firmware image, no resetcmd should be set in the environment variable.

**Furthermore this script is out-of-scope for this PR as I've opened up a separate PR for this.
Stealing my code to this PR is very disrespectful.**

grosjo replied:
@CodeFetch Your code just is not working. Please do keep polite and avoid bold  and nervous comments, this is totally not usefull

Removing the resetcmd here makes it that there is no resetcmd nowhere. The resetcmd must be initiated somewhere. So if not here, then where ?

grosjo replied:
I sourced a yuncire 830, which has UART

This is the error I get :

```
Using eth0 device
TFTP from server 192.168.0.141; our IP address is 192.168.0.144
Filename 'upgrade.bin'.
Load address: 0x80060000
Loading: #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #############
done
Bytes transferred = 4391347 (4301b3 hex)
firmware resetcmd string: YUNCOREsetenv bootcmd "bootm 0x9f050000 || bootm 0x9fe80000" &&  setenv resetcmd "erase 0x9f050000 +0xfa0000 
Please wait......
syntax error

NOTIFICATION: firmware resetcmd FINISHED
Rebooting......

```

CodeFetch replied:
@grosjo Please upload the resulting firmware image so I can have a look.

CodeFetch replied:
And please try the version from my PR without your changes...

CodeFetch replied:
I've found your problem... Look at the dd-line. The command exceeds 192 characters with your addition of the resetcmd (221). Thus it gets cut.

https://github.com/openwrt/openwrt/pull/2362/files#diff-f98caa543f5e6dc60bf413ccf8117b69R6

> Removing the resetcmd here makes it that there is no resetcmd nowhere. The resetcmd must be initiated somewhere. So if not here, then where ?

The resetcmd is temporarily defined by the command string at the beginning of the image.

YunCore's U-Boot TFTP  recovery works like this:
1. Load image via TFTP
2. Check whether there is the magic string "YUNCORE" at the beginning of the image
	If yes:
		Execute following command up to a NULL byte
	If no:
		Check whether resetcmd environment variable is set
		If yes:
			Execute command set in resetcmd environment variable
		If no:
			Print error message

Thus you should not set the resetcmd environment variable, because once you've done that the router will accept any image via TFTP and execute the command set in the variable every time...

grosjo replied:
No, you are absolutely wrong

I removed the so called 'keyword'  YUNCORE and then this solve this first error.
However, there is a second error (see below)

```
TFTP from server 192.168.0.141; our IP address is 192.168.0.144
Filename 'upgrade.bin'.
Load address: 0x80060000
Loading: #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #############
done
Bytes transferred = 4391347 (4301b3 hex)
firmware resetcmd string: setenv bootcmd "bootm 0x9f050000 || bootm 0x9fe80000" &&  saveenv && erase 0x9f050000 +0xfa0000 && cp.b 0x80 
no reset button command in firmware
checking the reset button cmd in env
env resetcmd string: erase 0x9f050000 +0xFA0000&&cp.b 0x80060000 0x9f050000 $filesize 
Please wait......
Erasing flash... 
First 0x5 last 0xfe sector size 0x10000
 254
Erased 250 sectors
Copy to Flash... write addr: 9f050000
done

NOTIFICATION: env resetcmd FINISHED
Rebooting......

eth1 up
eth0, eth1
Setting 0x181162c0 to 0x409a100
Hit any key to stop autoboot:  0 
## Booting image at 9fe80000 ...
Bad Magic Number
## Booting image at 9f050000 ...
Bad Magic Number

```

grosjo replied:
It is now working fine

grosjo commented on target/linux/ath79/dts/qca9531_yuncore_cpe830.dts:
```diff
@@ -0,0 +1,122 @@
+
+		wlan {
+			label = "cpe830:green:wlan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
```
done

grosjo commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -275,6 +275,16 @@ yuncore,a770)
+yuncore,cpe830)
+	ucidef_set_led_netdev "lan" "LAN" "$boardname:green:lan" "eth0"
+	ucidef_set_led_netdev "wan" "WAN" "$boardname:green:wan" "eth1"
+	ucidef_set_led_netdev "wlan" "WLAN" "$boardname:green:wlan" "wlan0"
```
so what should I put here in 01_leds ?

adrianschmutzler replied:
You can just remove the "wlan" line here.

grosjo replied:
ok done

grosjo commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -275,6 +275,16 @@ yuncore,a770)
 	ucidef_set_led_switch "lan" "LAN" "$boardname:green:lan" "switch0" "0x10"
 	;;
+yuncore,cpe830)
+	ucidef_set_led_netdev "lan" "LAN" "$boardname:green:lan" "eth0"
```
then why does it work on my device ?

grosjo replied:
I put 
`ucidef_set_led_switch "lan" "LAN" "$boardname:green:lan" "switch0"
`ok for you ?

grosjo replied:
@981213 ok for you ?

981213 replied:
> then why does it work on my device ?

a netdev trigger on eth0  results in a always-on lan led (even when there's no cable connected.)

grosjo replied:
discussion coninues below

981213 commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -275,6 +275,15 @@ yuncore,a770)
 	;;
+yuncore,cpe830)
+	ucidef_set_led_netdev "wan" "WAN" "$boardname:green:wan" "eth1"
+	ucidef_set_led_switch "lan" "LAN" "$boardname:green:lan" "switch0"
```
That's not how it works. ucidef_set_led_switch needs a 5th argument which is a port mask. You could connect a network cable to the lan port on this router and execute the following command on your router:
```
swconfig dev switch0 show
```
and find out which of port1-port4 is link up.
If it's port 1, the port mask here should be 0x02, port2->0x04 port3-> 0x08 port4->0x10 respectively.

grosjo replied:
# swconfig dev switch0 show
Global attributes:
	enable_vlan: 0
	ar8xxx_mib_poll_interval: 500
	ar8xxx_mib_type: 0
	enable_mirror_rx: 0
	enable_mirror_tx: 0
	mirror_monitor_port: 0
	mirror_source_port: 0
	arl_table: address resolution table
Port 0: MAC 44:d1:fa:75:20:65
Port 4: MAC 00:0e:c6:a4:84:38

Port 0:
	mib: MIB counters
RxGoodByte  : 290836 (284.0 KiB)
TxByte      : 62872 (61.3 KiB)

	pvid: 0
	link: port:0 link:up speed:1000baseT full-duplex txflow rxflow 
Port 1:
	mib: No MIB data
	pvid: 0
	link: port:1 link:down
Port 2:
	mib: No MIB data
	pvid: 0
	link: port:2 link:down
Port 3:
	mib: No MIB data
	pvid: 0
	link: port:3 link:down
Port 4:
	mib: MIB counters
RxGoodByte  : 63148 (61.6 KiB)
TxByte      : 292840 (285.9 KiB)

	pvid: 0
	link: port:4 link:up speed:100baseT full-duplex auto

CodeFetch commented:
@grosjo Now that you have both CPE830 and CPE830D can you please check the LED GPIOs?

psyborg55 commented:
> Saying it's invalid again and again without providing a reason isn't helpful at all.

 it's invalid again and again for the obvious reason.

is it more helpful now?

981213 commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -275,6 +275,15 @@ yuncore,a770)
 	;;
+yuncore,cpe830)
+	ucidef_set_led_netdev "wan" "WAN" "$boardname:green:wan" "eth1"
+	ucidef_set_led_switch "lan" "LAN" "$boardname:green:lan" "switch0" "0x00"
```
What's this 0x00 for? a led that blinks for no port at all?

grosjo replied:
Sorry, typing mistake. I fixed it.
Leds are not really a big concern, as this is an outdoor device (will one really go to check the leds in the middle of a typhoon ?)

981213 commented:
You should do some testing before requesting a review.
Anyway I'm tired of dealing with a PR whose author always keep silent about what's broken.

981213 commented:
To save you some effort for creating a 4th PR I'll reopen this one.

For future reviewers:
The author never explained why the 1st PR contains a different LED GPIO definition. This PR could be merged if someone else can verify it on both CPE830 and CPE830D or remove the claimed (D) suffix and cross-check its support with CPE830 code in ar71xx target.

grosjo commented:
Device is working now very fined on my installation (bootlog below), and thus for months now with no down time.

```
[    0.000000] Linux version 4.19.79 (admin@gjserver) (gcc version 8.3.0 (OpenWrt GCC 8.3.0 r11266-34939711a5)) #0 Sat Oct 19 11:16:57 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
[    0.000000] MIPS: machine is YunCore/KuWfi CPE830(D)
[    0.000000] SoC: Qualcomm Atheros QCA9533 ver 2 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000]   Normal zone: 128 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from 0x8049a730 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 58824K/65536K available (3667K kernel code, 149K rwdata, 504K rodata, 1176K init, 195K bss, 6712K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 650.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5880801374 ns
[    0.000008] sched_clock: 32 bits at 325MHz, resolution 3ns, wraps every 6607641598ns
[    0.008233] Calibrating delay loop... 432.53 BogoMIPS (lpj=2162688)
[    0.074858] pid_max: default: 32768 minimum: 301
[    0.079928] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086897] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.096474] devtmpfs: initialized
[    0.102057] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.112476] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.118976] pinctrl core: initialized pinctrl subsystem
[    0.125250] NET: Registered protocol family 16
[    0.151401] clocksource: Switched to clocksource MIPS
[    0.160433] NET: Registered protocol family 2
[    0.165813] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.173969] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.181318] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.188035] TCP: Hash tables configured (established 1024 bind 1024)
[    0.194888] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.201056] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.208014] NET: Registered protocol family 1
[    0.212678] PCI: CLS 0 bytes, default 32
[    0.215709] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.228192] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.234400] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.254155] io scheduler noop registered
[    0.258290] io scheduler deadline registered (default)
[    0.265979] pinctrl-single 1804002c.pinmux: 576 pins, size 72
[    0.274964] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.294677] m25p80 spi0.0: w25q128 (16384 Kbytes)
[    0.299681] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.306417] Creating 4 MTD partitions on "spi0.0":
[    0.311487] 0x000000000000-0x000000040000 : "u-boot"
[    0.318132] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.325606] 0x000000050000-0x000000ff0000 : "firmware"
[    0.333318] 2 uimage-fw partitions found on MTD device firmware
[    0.339560] Creating 2 MTD partitions on "firmware":
[    0.344861] 0x000000000000-0x000000170000 : "kernel"
[    0.351864] 0x000000170000-0x000000fa0000 : "rootfs"
[    0.358833] mtd: device 4 (rootfs) set to be root filesystem
[    0.365492] 1 squashfs-split partitions found on MTD device rootfs
[    0.372078] 0x000000430000-0x000000fa0000 : "rootfs_data"
[    0.379142] 0x000000ff0000-0x000001000000 : "art"
[    0.387024] libphy: Fixed MDIO Bus: probed
[    0.742496] ag71xx 19000000.eth: Could not connect to PHY device. Deferring probe.
[    1.421809] libphy: ag71xx_mdio: probed
[    1.427233] libphy: ar8xxx-mdio: probed
[    1.440079] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.492898] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.502971] eth0: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.512595] NET: Registered protocol family 17
[    1.517341] 8021q: 802.1Q VLAN Support v1.8
[    1.863666] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.874303] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    1.881732] Warning: unable to open an initial console.
[    1.896006] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.909465] devtmpfs: mounted
[    1.918995] Freeing unused kernel memory: 1176K
[    1.923795] This architecture does not have kernel memory protection.
[    1.930575] Run /sbin/init as init process
[    2.561407] random: fast init done
[    2.886406] init: Console is alive
[    2.890220] init: - watchdog -
[    4.034324] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.215590] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.233688] init: - preinit -
[    5.503763] random: jshn: uninitialized urandom read (4 bytes read)
[    5.561936] random: jshn: uninitialized urandom read (4 bytes read)
[    5.655116] random: jshn: uninitialized urandom read (4 bytes read)
[    7.382672] eth0: link up (1000Mbps/Full duplex)
[    8.694730] jffs2: notice: (804) jffs2_build_xattr_subsystem: complete building xattr subsystem, 3 of xdatum (2 unchecked, 1 orphan) and 9 of xref (1 dead, 0 orphan) found.
[    8.713084] mount_root: switching to jffs2 overlay
[    8.739959] overlayfs: upper fs does not support tmpfile.
[    8.755399] urandom-seed: Seeding with /etc/urandom.seed
[    8.953842] eth0: link down
[    8.968347] procd: - early -
[    8.972182] procd: - watchdog -
[    9.645113] procd: - watchdog -
[    9.648617] procd: failed to set stdio: No such device
[    9.654173] procd: - ubus -
[    9.940841] urandom_read: 6 callbacks suppressed
[    9.940849] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.955740] procd: - init -
[   10.656434] kmodloader: loading kernel modules from /etc/modules.d/*
[   10.712956] loop: module loaded
[   10.719273] Loading modules backported from Linux version v5.4-rc2-0-gda0c9ea146cb
[   10.727320] Backport generated by backports.git v5.4-rc2-1-0-g5a0c323a
[   10.785609] xt_time: kernel timezone is -0000
[   10.880986] urngd: v1.0.0 started.
[   10.998099] ath: EEPROM regdomain: 0x0
[   10.998108] ath: EEPROM indicates default country code should be used
[   10.998111] ath: doing EEPROM country->regdmn map search
[   10.998125] ath: country maps to regdmn code: 0x3a
[   10.998131] ath: Country alpha2 being used: US
[   10.998134] ath: Regpair used: 0x3a
[   11.010518] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   11.011553] ieee80211 phy0: Atheros AR9531 Rev:2 mem=0xb8100000, irq=11
[   11.101606] kmodloader: done loading kernel modules from /etc/modules.d/*
[   11.446866] random: crng init done
[   20.213712] br-lan: port 1(eth0) entered blocking state
[   20.219227] br-lan: port 1(eth0) entered disabled state
[   20.225079] device eth0 entered promiscuous mode
[   20.232360] eth0: link up (1000Mbps/Full duplex)
```

grosjo commented:
> You should at least do some testing before requesting a review.
> Anyway I'm tired of dealing with a PR whose author always keep silent about what's broken.

There is nothing broken, the only pending is for someone to push it through

grosjo commented:
@981213 What "changes" are you requesting ?

CodeFetch commented:
@grosjo Several people asked you to check the GPIO definitions on real hardware, because you changed them without testing. 

Imagine you are a car mechanic and your fellow asks you to review a brake line he repaired. Now you see a fluid leaking from the line and ask him: "Why is there fluid leaking?". He takes a handkerchief, wipes the fluid away and says: "Which fluid are you talking about?". That's how it appears to me when you ask which changes are being requested.

grosjo commented:
@981213, @adrianschmutzler, how to move forward on this ?

adrianschmutzler commented:
Well, the last comment (inaccurately marked off-topic by someone) told you that you should finally retrieve the correct GPIO definition on your actual hardware, as several other have requested in the past, too.

Despite, you should have realized now that your PRs have been rejected and are just kept open to prevent you from posting them as new again.

grosjo committed:
ath79: add support for KuWfi CPE830(D) / YunCore CPE830(D)

Board uses ath9k driver (and not ath10k as previously stated in Openwrt)

Short specification:

    650/600/216 MHz (CPU/DDR/AHB)
    2x 10/100 Mbps Ethernet, passive PoE support
    64 MB of RAM (DDR2)
    16 MB of FLASH
    2T2R 2.4 GHz with external PA, up to 30 dBm (1000mW)
    2x internal 14 dBi antennas
    8x LED, 1x button
    No UART on PCB on some versions
    Display panel with 2x buttons (F/N) not supported (and not relevant in OpenWrt)

Flash instructions

    Connect PC with 192.168.0.141 to WAN port
    Install a TFTP server on your PC ('atftp' is doing the job for instance)
    Copy your firmware in the TFTP folder as upgrade.bin
    Power up device pushing the 'reset' button
    The device shall upload upgrade.bin, install it and reboot
    Device shall be booting on 192.168.1.1 as default

Signed-off by: Joan Moreau jom@grosjo.net

grosjo commented:
> Well, the last comment (inaccurately marked off-topic by someone) told you that you should finally retrieve the correct GPIO definition on your actual hardware, as several other have requested in the past, too.
> 
> Despite, you should have realized now that your PRs have been rejected and are just kept open to prevent you from posting them as new again.

This is really bad behavior. My devices are running well with this patch, you are blaming the wind, but can you be rather constructive ?

grosjo commented:
> Well, the last comment (inaccurately marked off-topic by someone) told you that you should finally retrieve the correct GPIO definition on your actual hardware, as several other have requested in the past, too.
> 
> Despite, you should have realized now that your PRs have been rejected and are just kept open to prevent you from posting them as new again.

This is really bad behavior. 
My devices are running well with this patch, you are blaming the wind, but can you be rather constructive ?

grosjo commented:
> Well, the last comment (inaccurately marked off-topic by someone) told you that you should finally retrieve the correct GPIO definition on your actual hardware, as several other have requested in the past, too.
> 
> Despite, you should have realized now that your PRs have been rejected and are just kept open to prevent you from posting them as new again.

This is really bad behavior. 
My devices are running well with this patch, you are blaming the wind, but can you be rather constructive ?

CodeFetch commented:
> My devices are running well with this patch, you are blaming the wind, but can you be rather constructive ?

Just test all the GPIOs on real hardware. This discussion leads nowhere.

adrianschmutzler commented:
> > Well, the last comment (inaccurately marked off-topic by someone) told you that you should finally retrieve the correct GPIO definition on your actual hardware, as several other have requested in the past, too.
> > Despite, you should have realized now that your PRs have been rejected and are just kept open to prevent you from posting them as new again.
> 
> This is really bad behavior.
> My devices are running well with this patch, you are blaming the wind, but can you be rather constructive ?

You have received constructive feedback from three core-developers (more than most of the PRs get) and you have either ignored them or tried to deceive us.

From my point of view, there is no trust left. So even if you finally updated the GPIOs, I would not take it anymore. I'm done with you, and I'm very sorry that I'm forced to say it that harshly, as you do not seem to understand it in a nicer way.

grosjo commented:
Never tried to "deceive" anyone. This os working perfectly well, behavior of people here however is really broken

adrianschmutzler commented:
After 8 months, I think it's rather obvious that this won't be merged anymore.
I'm closing this now, as there is no point in having it lying around forever.

grosjo commented:
THis is really nut. You shall really get back to normal behavior

adrianschmutzler commented:
I regularly close old PRs that have not seen any activity for a long time, and yours now has reached half a year of inactivity.

Neustradamus commented:
After several years, it has been merged in:
- https://github.com/openwrt/openwrt/pull/17279
